### PR TITLE
Fetch and store openId configs

### PR DESCRIPTION
**Pull request by robmckinnon**
- Add updateOpenIdConfigs() function to fetch and store configs.
- Fetch and store openIdConfig for each authorisation server without openIdConfig.
- Trigger async update of openIdConfigs when client requests ASPSP list (for now this seems a good time to fetch them)
- Await db cleanup to finish in test setups and teardowns

For: REFAPP-45

### CODE REVIEW

File: app/ob-directory.js [hunk 1]
```diff
@@ -113,10 +113,11 @@ const OBAccountPaymentServiceProviders = async (req, res) => {
   res.setHeader('Access-Control-Allow-Origin', '*');
```
Comment (ezodude) [inline app/ob-directory.js]: Perhaps I'm slightly confused, but the code in L114 - L122 feels slightly odd.

`let servers = authorisationServersForClient();` - retrieves the cached authorisation servers. If the cached authorisation servers exist, then they are updated.
__HOWEVER__ the updated list is never returned to the client.

Instead, the updated cached authorisation servers list is returned on a subsequent call.

File: app/authorisation-servers/authorisation-servers.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 // const log = require('debug')('log');
+const debug = require('debug')('debug');
 const error = require('debug')('error');
 const { getAll, get, set } = require('../storage');
+const { getOpenIdConfig } = require('./openid-config');
 
 const AUTH_SERVER_COLLECTION = 'aspspAuthorisationServers';
```
[the remark this answers is not in the document]
Comment (robmckinnon) [reply app/authorisation-servers/authorisation-servers.js]: I'll change in next PR.

File: app/ob-directory.js (hunk 1 above)
Comment (robmckinnon) [reply app/ob-directory.js]: This was intended to be a mechanism to keep refreshing the list, but without causing a delay for the web client.

Also spotted bug, line 113 should be:
```
let servers = await authorisationServersForClient();
```

I'll fix line 113 in next PR.

Welcome suggestions for how keeping the list updated could be made more realistic. Maybe a story for a future sprint.

Comment (ezodude) [reply app/ob-directory.js]: Makes sense.

**Pull request merged**